documentation/writing_tests.md

```diff
@@ -206,20 +243,168 @@ The pictures:
 
 Mocha's `before` and `beforeEach` are methods that allow you to specify functions that should be executed either before all of the tests run or before each test runs, respectively. Similarly, `after` and `afterEach` exist so that developers can specify functions that should be executed after all tests run or after each test runs, respectively.
 
-We recommend using `beforeEach` rather than `before`, just as much as we recommend using `afterEach` rather than `after`. The idea is that each one of your test cases should not depend on a state that is shared with other tests. Use `beforeEach` to execute tasks that will prepare the resources needed for each test to run cleanly, and `afterEach` to tear down or clean those settings before the next test runs.
+We recommend using `beforeEach` rather than `before`, just as much as we recommend using `afterEach` rather than `after`. The idea is that each test case should not depend on a state that is shared with other tests. Use `beforeEach` to execute tasks that will prepare the resources needed for each test to run cleanly, and `afterEach` to tear down or clean those settings before the next test runs.
+
+`before` and `after` can be used to define or create heavy resources that could cleanly be used by more than one test. We will see some examples below.
+
+A discouraged example of `before` and `after` ([source](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/template/template/test/internal/beforeAfter.spec.ts#L12)):
+
+```ts
+describe.skip("Discouraged example of `before` and `after`", function() {
```

Comment:
skip ?

Reply:
This is discouraged! I don't want something discouraged to run 😁

Reply:
I've removed the skip here!